kernel/task/src/lib.rs

```diff
@@ -206,6 +206,10 @@ pub enum RunState {
     Runnable,
     /// blocked on something, like I/O or a wait event
     Blocked,
+    /// The `Task` has been suspended from proceding further.
+    ///
+    /// Currently, this is only triggered by a Ctrl + Z in the terminal.
+    Suspended,
```

Comment:
There may be other reasons Linux has a separate state but I can't think of any.

Reply:
I see. So we either need a separate runstate or a "blocked semaphore" that counts how many times a task has been blocked (kind of like how we do preemption holds). 

Q: even with a separate runstate, consider how the following scenario plays out:
1. task is running as normal
2. task blocks itself upon something, e.g., waiting for I/O. 
    --> task is now Blocked
3. user suspends the task using the shell
    --> task is now Suspended (?)
4. the I/O operation completes, unblocking the task
   --> now what should happen? 

The task is no longer blocked but it's also not runnable... we'd need a sort of "stack" to track prior runstate transitions to accomplish this.

Alternatively, should "Suspended" be a completely separate bit of information rather than a runstate? That's potentially even more annoying to handle, but it's also more correct.

Reply:
I added a separate `suspended` field to `Task` in 2f0d95c.